### PR TITLE
personal-details hookup

### PR DESCRIPTION
**Pull request by lewiskyron**
## Description
This PR makes the personal details page functional. I have removed the form from the `app` directory and move it as a component to the `components folder`. This has solved issue https://github.com/ucey-star/ID8/issues/27 and now the personal-details page does not use the 'use client' as it did before. Additionally this PR closes #issue https://github.com/ucey-star/ID8/issues/35

How the changes work:
- We get the user from the page component and pass it as a prop to the personal details form. From this point we make a post request with the user information and redirect them to the home page.

Some changes:
- In the `personal-details` page I have removed the `name` and  `email` since we already get this from Google authentication. Consequently, I have changed the `name` to be `username` and the `age` to be `Date of Birth` since we want to dynamically update our users age on their birthday's. Currently, we have a database function that updates the user's ages everyday ensuring we also have accurate user ages.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Tests (adding or updating tests)
- [ ] Documentation update

## Testing Instructions
- checkout into the branch 
- run `npm install` we have some new dependencies installed.
- Go the personal-details page input some information and if the post request is a a success you should be redirected to the home page. 

## Screenshots (if applicable)

If your changes include visual components, add screenshots showing the affected pages or elements.

### CODE REVIEW

File: src/components/PersonalDetailsForm.tsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+						display: "flex",
+						flexDirection: "column",
+						gap: "40px",
+						marginBottom: "32px",
+						padding: "8px",
+					}}
+				>
+					<TextField
+						fullWidth
+						label="username"
```
Comment (FlaIespa) [inline src/components/PersonalDetailsForm.tsx]: I would capitalize the word!

File: src/app/personal-details/page.tsx [hunk 2]
```diff
@@ -76,146 +60,9 @@ const PersonalDetailsScreen: React.FC = () => {
...
-						value={workplace}
-						onChange={(e) => setWorkplace(e.target.value)}
-						variant="outlined"
-					/>
-
-					{/* LinkedIn URL Field */}
-					<TextField
-						fullWidth
-						label="LinkedIn URL*"
-						value={linkedin}
```
Comment (FlaIespa) [inline src/app/personal-details/page.tsx:202]: I don't believe that the asterisk here is necessary.

**Review by FlaIespa — APPROVED** [review]
Great work here, Kyron! I suggested a few changes for better UI. The work looks great to me.

File: src/app/personal-details/page.tsx (hunk 2 above)
Comment (lewiskyron) [reply src/app/personal-details/page.tsx:202]: Good catch! Makes the changes. Thanks

**Comment by lewiskyron** [comment]
Merging this into. Main I cannot see the vercel errors since I am not added as collaborator. The code works as expected and the deployment error will be addressed in another PR.

**Pull request merged**